yarn.lock

```diff
@@ -4517,7 +4522,7 @@ componentsjs-generator@^3.0.5:
     rdf-object "^1.13.1"
     semver "^7.3.2"
 
-componentsjs@^5.0.1, componentsjs@^5.2.0:
+componentsjs@^5.0.1:
```

Comment:
This seems odd. I would expect this one to be done in favor of `componentsjs@^5.3.1`.
I see some other libs also have some duplicates.

When running this update command, should we also enforce a reset of `yarn.lock`?

Reply:
The problem with a reset of the `yarn.lock` file is then these errors appear (in the case of saxes its because multiple major versions of that package are required by nested dependencies; the last error I haven't diagnosed)

```bash
node_modules/saxes/saxes.d.ts:46:71 - error TS2344: Type 'O' does not satisfy the constraint 'SaxesOptions'.
  Type 'O' is not assignable to type 'CommonOptions'.

46 export declare type OpenTagStartHandler<O> = (tag: StartTagForOptions<O>) => void;
                                                                         ~

  node_modules/saxes/saxes.d.ts:46:41
    46 export declare type OpenTagStartHandler<O> = (tag: StartTagForOptions<O>) => void;
                                               ~
    This type parameter might need an `extends CommonOptions` constraint.
  node_modules/saxes/saxes.d.ts:46:41
    46 export declare type OpenTagStartHandler<O> = (tag: StartTagForOptions<O>) => void;
                                               ~
    This type parameter might need an `extends SaxesOptions` constraint.

node_modules/saxes/saxes.d.ts:55:80 - error TS2344: Type 'O' does not satisfy the constraint 'SaxesOptions'.
  Type 'O' is not assignable to type 'CommonOptions'.

55 export declare type AttributeHandler<O> = (attribute: AttributeEventForOptions<O>) => void;
                                                                                  ~

  node_modules/saxes/saxes.d.ts:55:38
    55 export declare type AttributeHandler<O> = (attribute: AttributeEventForOptions<O>) => void;
                                            ~
    This type parameter might need an `extends CommonOptions` constraint.
  node_modules/saxes/saxes.d.ts:55:38
    55 export declare type AttributeHandler<O> = (attribute: AttributeEventForOptions<O>) => void;
                                            ~
    This type parameter might need an `extends SaxesOptions` constraint.

node_modules/saxes/saxes.d.ts:62:61 - error TS2344: Type 'O' does not satisfy the constraint 'SaxesOptions'.
  Type 'O' is not assignable to type 'CommonOptions'.

62 export declare type OpenTagHandler<O> = (tag: TagForOptions<O>) => void;
                                                               ~

  node_modules/saxes/saxes.d.ts:62:36
    62 export declare type OpenTagHandler<O> = (tag: TagForOptions<O>) => void;
                                          ~
    This type parameter might need an `extends CommonOptions` constraint.
  node_modules/saxes/saxes.d.ts:62:36
    62 export declare type OpenTagHandler<O> = (tag: TagForOptions<O>) => void;
                                          ~
    This type parameter might need an `extends SaxesOptions` constraint.

node_modules/saxes/saxes.d.ts:69:62 - error TS2344: Type 'O' does not satisfy the constraint 'SaxesOptions'.
  Type 'O' is not assignable to type 'CommonOptions'.

69 export declare type CloseTagHandler<O> = (tag: TagForOptions<O>) => void;
                                                                ~

  node_modules/saxes/saxes.d.ts:69:37
    69 export declare type CloseTagHandler<O> = (tag: TagForOptions<O>) => void;
                                           ~
    This type parameter might need an `extends CommonOptions` constraint.
  node_modules/saxes/saxes.d.ts:69:37
    69 export declare type CloseTagHandler<O> = (tag: TagForOptions<O>) => void;
                                           ~
    This type parameter might need an `extends SaxesOptions` constraint.

packages/actor-init-query/lib/QueryEngineBase.ts:63:7 - error TS2322: Type 'boolean | void | BindingsStream | (AsyncIterator<Quad> & ResultStream<Quad>)' is not assignable to type 'ReturnType<QueryTypeOut["execute"]>'.
  Type 'boolean' is not assignable to type 'ReturnType<QueryTypeOut["execute"]>'.

63       return result.execute();
         ~~~~~~~~~~~~~~~~~~~~~~~~


Found 5 errors in 2 files.

Errors  Files
     4  node_modules/saxes/saxes.d.ts:46
     1  packages/actor-init-query/lib/QueryEngineBase.ts:63
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Reply:
Hmm, haven't seen those myself before. Must be due to some recent type changes (possible be external to comunica).
Since resetting the lockfile will only cause in-range updates, other people installing comunica now will probably run into the same issues, so this seems quite problematic.

(we used to have greenkeeper notifying us of these things, but that project has gone dead unfortunately)